sdks/java/extensions/google-cloud-platform-core/src/test/java/org/apache/beam/sdk/util/RetryHttpRequestInitializerTest.java

```diff
@@ -255,6 +255,7 @@ public Integer answer(InvocationOnMock invocation) {
    * is invoked.
    */
   @Test
+  @SuppressWarnings("AssertionFailureIgnored")
```

Comment:
Can this test be improved such that we don't need this suppression? The [ErrorProne AssertionFailureIgnored](https://errorprone.info/bugpattern/AssertionFailureIgnored) page suggests using JUnit's `assertThrows()`.

Reply:
I did not find an easy way to do this and also the proposed solution by errorprone is invalid. See https://github.com/google/error-prone/issues/1059
Any suggestions are welcome.

Reply:
You could re-create the idea behind `assertThrows` with:

```java
Throwable failure = null;
try {
  getRequest.execute();
} catch (Throwable e) {
  failure = e;
}
assertNotNull("Expected execute to throw an exception", failure);
assertThat(failure, Matchers.instanceOf(SocketTimeoutException.class));
assertEquals(1 + defaultNumberOfRetries, executeCount.get());
expectedLogs.verifyWarn("performed 10 retries due to IOExceptions");
```

Reply:
I don't know why I was thinking in a way more complex idea. Thanks, added this.